cosmos/dbt/project.py

```diff
@@ -34,12 +36,36 @@ def has_non_empty_dependencies_file(project_path: Path) -> bool:
     return False
 
 
+def copy_dbt_packages(source_folder: Path, target_folder: Path) -> None:
+    """
+    Copies the dbt packages related files and directories from source_folder to target_folder.
+
+    :param: source_folder: The base directory where paths are sourced from.
+    :param: target_folder: The directory where paths will be copied to.
+    """
+    logger.info("Copying dbt packages to temporary folder...")
+    dbt_packages_paths = [DBT_PACKAGES_FOLDER, PACKAGE_LOCKFILE_YML]
+
+    for relative_path in dbt_packages_paths:
+        src_path = source_folder / relative_path
+        dst_path = target_folder / relative_path
+
+        os.makedirs(os.path.dirname(dst_path), exist_ok=True)
+
+        if src_path.is_dir():
+            shutil.copytree(src_path, dst_path, dirs_exist_ok=True)
+        else:
+            shutil.copy(src_path, dst_path)
```

Comment:
```suggestion
            shutil.copy2(src_path, dst_path)
```

Reply:
Reverting due to failing test: https://github.com/astronomer/astronomer-cosmos/actions/runs/14494584744/job/40658978949?pr=1668

`copy2` handles the scenario where the original file doesn't exist - something that we care about